x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_indices/indices_logic.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { Meta } from '../../../../../common/types';
+import { HttpError } from '../../../../../common/types/api';
+import { DEFAULT_META } from '../../../shared/constants';
+import { flashAPIErrors, clearFlashMessages } from '../../../shared/flash_messages';
+import { updateMetaPageIndex } from '../../../shared/table_pagination';
+import { IndicesAPILogic } from '../../logic/indices_api';
+import { SearchIndex } from '../../types';
+
+export interface IndicesActions {
+  makeRequest: typeof IndicesAPILogic.actions.makeRequest;
+  apiSuccess({ indices, meta }: { indices: SearchIndex[]; meta: Meta }): {
+    indices: SearchIndex[];
+    meta: Meta;
+  };
+  apiError(error: HttpError): HttpError;
+  onPaginate(newPageIndex: number): { newPageIndex: number };
+}
+export interface IndicesValues {
+  indices: SearchIndex[];
+  meta: Meta;
+}
+
+export const IndicesLogic = kea<MakeLogicType<IndicesValues, IndicesActions>>({
+  connect: {
+    actions: [IndicesAPILogic, ['makeRequest', 'apiSuccess', 'apiError']],
+  },
+  path: ['enterprise_search', 'content', 'indices_logic'],
+  actions: {
+    onPaginate: (newPageIndex) => ({ newPageIndex }),
+  },
+  reducers: () => ({
+    indices: [
+      [],
+      {
+        apiSuccess: (_, { indices }) => indices,
```

Comment:
suggestion: grabbing the value from the apiLogic is a little cleaner.

Reply:
Two reasons to use selectors btw:
1. They're more performant
2. Rather than duplicating data everywhere and potentially missing some flows (like you almost did with the loading state here), you're guaranteed to have one source of truth for the data you're using.

Reply:
Oh yeah, makes sense :D will do the changes and update the PR.